### PR TITLE
Added dockerfile

### LINKED ISSUE #13: Add docker file
Add a docker file to create an image of this repository. Refer [this](https://docs.docker.com/engine/reference/builder/).

### PR DESCRIPTION
**Pull request by poohcom1**
# Added dockerfile
Resolves #13 

## Development
Docker-compose automates volume mounting, so hot-reloading would still work in the dockerized environment. Both container and host port is 3000 by default.

- Added `Dockerfile`
- Added `docker-compose.yml`
- Added `.dockerignore`

Building:
```sh
docker-compose up -d
```

## Production

 - Added `Dockerfile.prod`
 - Added `nginx/nginx.conf`

Building:
```sh
docker build . -f Dockerfile.prod
```

* Note: I realized that `package-lock.json` is not being tracked, is there a reason for this? To my knowledge the lock file is required for proper source control, so it's usually also added to the dockerfile image. But since this project isn't tracking it, I've left it out.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM node:14.17.3-alpine
+
+ENV NODE_ENV=development
+
+WORKDIR /app
+
+COPY package.json ./
```
Comment (Anurag5086) [inline Dockerfile]: We can use this instead, as we will also be tracking the lock file as well.
```suggestion
COPY package*.json ./
```

File: Dockerfile.prod [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM node:14.17.3-alpine as build
+
+ENV NODE_ENV=production
+
+WORKDIR /app
+
+COPY package.json ./
```
Comment (Anurag5086) [inline Dockerfile.prod:7]: Same goes here.

**Review by Anurag5086 — CHANGES_REQUESTED** [review]

**Comment by Anurag5086** [comment]
@poohcom1 Could you also add the Docker setup for dev and prod in the `README.md` file.

File: Dockerfile.prod (hunk 2 above)
Comment (Anurag5086) [inline Dockerfile.prod]:
```suggestion
COPY package*.json ./
```

**Review by Anurag5086 — CHANGES_REQUESTED** [review]

**Comment by poohcom1** [comment]
@Anurag5086 I've applied the requested changes and added docker build instructions in the readme

Edit: Seems like code quality check is failing, but it was due to the style conventions in README.md, which I tried to match the rest of the document. Should I change that? I can update the rest of the readme too if you want

**Comment by Anurag5086** [comment]
> @Anurag5086 I've applied the requested changes and added docker build instructions in the readme
> 
> Edit: Seems like code quality check is failing, but it was due to the style conventions in README.md, which I tried to match the rest of the document. Should I change that? I can update the rest of the readme too if you want

Sure, go ahead! 👍

**Comment by poohcom1** [comment]
@Anurag5086  Alright, I've fixed the formatting issue in the readme, although a lot of it did seem like stylistic choice so you might wanna tweak Codacy's setting since it seems like there's a lots of warnings currently haha.

**Review by Anurag5086 — APPROVED** [review]
Looks Good @poohcom1! 👍  
Thank you for your contributions. 💯 
You can check out other issues as well.

**Pull request merged**